.eleventy.js

```diff
@@ -295,6 +297,71 @@ module.exports = function (eleventyConfig) {
     },
   );
 
+  /*
+   * Shortcode to render demo dates
+   * Renders based on data file at ./src/_data/registerdemos.js
+   */
+  eleventyConfig.addPairedShortcode(
+    'registerForDemoLinks',
+    (children, locale) => {
+      const langStrings = {
+        en: {
+          enDemo: 'English demo',
+          frDemo: 'French demo',
+          timezone: 'Eastern time',
+        },
+        fr: {
+          enDemo: 'Démonstration en anglais',
+          frDemo: 'Démonstration en français',
+          timezone: 'HAE',
+        },
+      };
+
+      return `
+      <ul class="mb-300">
+        ${demoDates
+          .map(date => {
+            const options = {
+              weekday: 'long',
+              year: 'numeric',
+              month: 'long',
+              day: 'numeric',
+              timeZone: 'UTC',
+            };
+            const prnDt = new Date(date.date).toLocaleString(locale, options);
+
+            const datelink = `<gcds-link external href="${date.link}">${date.lang === 'en' ? langStrings[locale].enDemo : langStrings[locale].frDemo}</gcds-link>`;
+            const time = `<time>
+              ${
+                locale === 'en'
+                  ? `${prnDt}, ${convertTime(date.starttime)} - ${convertTime(date.endtime)} ${langStrings[locale].timezone}`
+                  : `${prnDt}, de ${date.starttime.replace(':', 'h')} à ${date.endtime.replace(':', 'h')} ${langStrings[locale].timezone}`
+              }
+            </time>`;
+
+            // Don't render link if date has passed
```

Comment:
If all the dates have passed, it will still render a ul but with no li. Suggestion:
Get all valid dates and put it in an array
If array is empty, show "No upcoming dates" (or, just leave blank so we don't need to wait for more content)
If array is not empty, proceed to display this code, and you can probably take out condition here since you now only have valid dates

Reply:
I like adding "No upcoming dates" if the array is empty, doesn't leave people wondering if they are missing something.